Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -198,6 +198,19 @@ set_option linter.uppercaseLean3 false in
 
 end UnitaryGroup
 
+section specialUnitaryGroup
+
+variable (n) (α : Type v) [CommRing α] [StarRing α] [Invertible (2 : α)] {A : Matrix n n α}
+
+/--`Matrix.specialUnitaryGroup` is the group of unitary `n` by `n` matrices where the determinant
+is 1-/
+abbrev specialUnitaryGroup := (MonoidHom.restrict detMonoidHom (unitaryGroup n α)).ker
+
+theorem mem_specialUnitaryGroup_iff  (h : A ∈ unitaryGroup n α):
+    ⟨A, h⟩ ∈ specialUnitaryGroup n α ↔ A.det = 1 := by rfl
```

Comment:
I wonder whether it would be better to have `specialUnitaryGroup` be a submonoid of `Matrix n n α` (like `unitaryGroup`), to avoid this nested subtype annoyance?

Reply:
How would you implement this?

Reply:
One option would be `unitaryGroup n α ⊓ MonoidHom.mker detMonoidHom`.

Reply:
How would I create an element with this approach?

Reply:
I haven't tested this, but probably `⟨A, h1, h2⟩` where `h1` is a proof that `A` is unitary and `h2` is a proof that `A` is special.

Reply:
Thank you very much